src/client/Microsoft.Identity.Client/Extensibility/AcquireTokenParameterBuilderExtensions.cs

```diff
@@ -21,7 +21,10 @@ public static class AcquireTokenParameterBuilderExtensions
             IDictionary<string, string> extraHttpHeaders)
             where T : AbstractAcquireTokenParameterBuilder<T>
         {
-            builder.CommonParameters.ExtraHttpHeaders = extraHttpHeaders;
+            foreach (var kvp in extraHttpHeaders)
```

Comment:
Re-read the email. Seems like its just ExtraParameters mentioned in the email. I think that sounds better since these are neither headers nor query parameter. I can add another API in this extension class `.WithExtraParameters`

Reply:
CC: @jmprieur @localden

Reply:
We haven't really decided on WithExtraParameters yet. Still discussing with @jmprieur and @localden  if this is the right approach for all 1p features.